packages/react-dom/src/__tests__/ReactDOM-test.js

```diff
@@ -525,4 +525,29 @@ describe('ReactDOM', () => {
         '    in App (at **)',
     ]);
   });
+
+  it('should not remove _reactRootContainer from a root if unmountComponentAtNode is follow by a render', () => {
+    const containerParent = document.createElement('div');
+    const containerApp = document.createElement('div');
+
+    function App1() {
+      return <div id="app1" />;
+    }
+    function App2() {
+      return <div id="app2" />;
+    }
+
+    function Parent() {
+      ReactDOM.render(<App1 />, containerApp);
+      ReactDOM.unmountComponentAtNode(containerApp);
+      ReactDOM.render(<App2 />, containerApp);
```

Comment:
I wouldn't say this is a supported way to use React. There shouldn't be side effects in the rendering phase.

Can you come up with a repro case that is more valid? e.g. maybe you can reproduce this with a class component when calling `render` and `unmountComponentAtNode` inside `componentDidMount` or another lifecycle?

Reply:
Hi @gaearon,
Didn't have time to investigate more, but does the test is okay now ? or do I miss something ?